docs/proposals/rewrite-client-ip.md

```diff
@@ -0,0 +1,296 @@
+  // This field is required if mode is set.
+  TrustedAddresses []string `json:"trustedAddresses,omitempty"`
+
+  // SetIPRecursively configures whether recursive search is used when selecting the client's address from the X-Forwarded-For header.
+  // It is used in conjunction with TrustedAddresses.
+  // If enabled, NGINX will recurse on the values in X-Forwarded-Header from the end of array to start of array and select the first untrusted IP.
+  // For example, if X-Forwarded-For is [11.11.11.11, 22.22.22.22, 55.55.55.1], and TrustedAddresses is set to 55.55.55.1/0, NGINX will rewrite the client IP to 22.22.22.22.
+  // If disabled, NGINX will select the IP at the end of the array. In the previous example, 55.55.55.1 would be selected.
+  //
+  // +optional
+  SetIPRecursively *bool `json:"setIPRecursively,omitempty"`
+}
+
+// RewriteClientIPModeType defines how NGINX Gateway Fabric will determine the client's original IP address.
+// +kubebuilder:validation:Enum=ProxyProtocol;XForwardedFor
+type RewriteClientIPModeType string
+
+const (
+  // RewriteClientIPModeProxyProtocol configures NGINX to accept PROXY protocol and set the client's IP address to the IP address in the PROXY protocol header.
+  // Sets the proxy_protocol parameter to the listen directive on all servers, and sets real_ip_header to proxy_protocol: https://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_header.
+  RewriteClientIPModeProxyProtocol RewriteClientIPModeType = "ProxyProtocol"
+
+  // RewriteClientIPModeXForwardedFor configures NGINX to set the client's IP address to the IP address in the X-Forwarded-For HTTP header.
+  // Sets real_ip_header to XForwardedFor: https://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_header.
+  RewriteClientIPModeXForwardedFor RewriteClientIPModeType = "XForwardedFor"
+)
+
+```
+
+The benefits of this API are:
+
+1. Allow users to easily configure the two most common methods for rewriting the client IP address.
+2. Express configuration in terms of use cases.
+3. Secure by default. By requiring that TrustedAddresses is set, users will have to explicitly enable addresses to trust.
+4. Require minimal knowledge of NGINX configuration while still correlating fields to NGINX directives or behavior.
+5. Allow for extension to support other methods of rewriting IP addresses in the future.
+
+### Validation
+
+The Go API above does not contain validation annotation. Annotations should be added to enforce the following rules.
+
+- If `mode` is set, then `trustedAddresses` is required.
+- `trustedAddresses` can have up to 16 CIDR blocks.
+- `trustedAddresses` must be in CIDR block notation.
+
+### Status
+
+Status is set on the GatewayClass, not the `NginxProxy` resource. If the `NginxProxy` is invalid, set the `Accepted` condition on the GatewayClass to `False` with the reason `InvalidParameters`. See [gateway settings proposal](gateway-settings.md#status) for more details on status.
```

Comment:
We don't actually set `Accepted` to `False` because this would create a cascade of denying all configuration under that class if an invalid parameter is introduced. Instead, we mark the parameters as invalid and only use defaults, while preserving the acceptance of the class.

Reply:
That's right. I read that section, I swear 😅 . I meant ResolvedRefs here. I'll update